### PR TITLE
Check all hostnames are lowercase

### LINKED ISSUE #399: precheck for hostname casing
Seems that calico version of kubespray do not like hostname with capital letter in it.
Add a precheck to fail before install

### PR DESCRIPTION
**Pull request by ballot-scality**
Due to a bug in Calico, the hostname must be composed of only lowercase
characters : https://github.com/projectcalico/calico/issues/1572

Fixes #399

### CODE REVIEW

**Comment by NicolasT** [comment]
@ballot-scality This should go in 1.0

File: roles/preflight_checks/action_plugins/validate_inventory.py [hunk 1]
```diff
@@ -109,6 +109,28 @@ def check_ansible_user_is_not_root(task_vars):
...
+    '''Ensure there is only lower case hostnames in the inventory
+
+    Kubespray set the hostname with lower case based on the inventory content
+    but this breaks calico, therefore we should enforce the usage of lower
+    letter only in the hostnames defined in the inventory
+    Kubnespray uses the `inventory_hostname` variable
+    '''
+
+    for host, hostvars in sorted(task_vars['hostvars'].items()):
+        if any([letter.isupper() for letter in host]):
```
Comment (NicolasT) [inline roles/preflight_checks/action_plugins/validate_inventory.py:135]: I think we should extend this and make sure there are no other nonsense characters in the name either. Basically `^[a-z0-9]([a-z0-9\-]*[a-z0-9])?$` or something along those lines ('Starts with a-z0-9, can have a-z0-9 or - in the middle, ends with a-z0-9')

**Comment by NicolasT** [comment]
@bert-e wait

**Comment by ballot-scality** [comment]
Miss click on my part I intended to merge in 1.0 (hence the diverge). Will fix tomorrow as well as the regexp.

File: roles/preflight_checks/action_plugins/validate_inventory.py (hunk 1 above)
Comment (ballot-scality) [reply roles/preflight_checks/action_plugins/validate_inventory.py:135]: Hum, what about unicode characters then ? Technically it's valid, but does calico supports that ? If we go that far, shouldn't we do a full validation of the hostname provided as being valid FQDN instead ?

**Comment by ballot-scality** [comment]
@bert-e reset

File: roles/preflight_checks/action_plugins/validate_inventory.py (hunk 1 above)
Comment (NicolasT) [reply roles/preflight_checks/action_plugins/validate_inventory.py:135]: I would rule out too 'particular' things for now.

**Comment by NicolasT** [comment]
@ballot-scality You want to implement the tests as part of this PR?

**Comment by ballot-scality** [comment]
Would like too but can do in another PR if you prefer

**Comment by NicolasT** [comment]
Nah here is fine. Guess it shouldn't take too long to get done.

**Review by NicolasT — APPROVED** [review]
That's *a lot* more code than I expected :wink: but LGTM.

**Comment by NicolasT** [comment]
@bert-e status

**Pull request merged**